File: app/src/main/java/com/ravisharma/playbackmusic/database/QueueSongsDao.java
package com.ravisharma.playbackmusic.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.ravisharma.playbackmusic.database.model.QueueSongs;

import java.util.List;

@Dao
public interface QueueSongsDao {

    @Insert
    void addSongs(QueueSongs songs);

    @Query("DELETE FROM queueSongs WHERE id=:id")
    void deleteFromQueue(long id);

    @Query("DELETE FROM queueSongs")
    void removeAll();

    @Query("SELECT * FROM queueSongs")
    List<QueueSongs> getQueueSongs();
}

File: app/src/main/java/com/ravisharma/playbackmusic/database/dao/QueueSongsDao.kt
package com.ravisharma.playbackmusic.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.ravisharma.playbackmusic.database.model.QueueSongs

@Dao
interface QueueSongsDao {
    @Insert
    fun addSongs(songs: QueueSongs)

    @Query("DELETE FROM queueSongs WHERE id=:id")
    fun deleteFromQueue(id: Long)

    @Query("DELETE FROM queueSongs")
    fun removeAll() : Int

    @get:Query("SELECT * FROM queueSongs")
    val queueSongs: List<QueueSongs>
}